WestLaw UK.js

```diff
@@ -0,0 +1,139 @@
+		return "statute";
+	}
+	else if (doc.title.includes("s.")) {
+		return "statuteSection";
+	}
+	else {
+		return false;
+	}
+}
+
+function doWeb(doc, url) {
+	const detectedType = detectWeb(doc, url);
+
+	switch (detectedType) {
+		case "case":
+			return scrapeCase(doc, url);
+		case "statute":
+			return scrapeStatute(doc, url);
+		case "statuteSection":
+			return scrapeStatuteSection(doc, url);
+		default:
+			return false;
+	}
+}
+
+function scrapeCase(doc, url) {
+	const translator = Zotero.loadTranslator('web');
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	translator.setHandler('itemDone', function (obj, item) {
+		const citationArray = parseCitationList(text(doc, "#co_docContentWhereReported", 0));
+
+		item.reporter = citationArray[3];
+		item.reporterVolume = citationArray[2];
+		item.firstPage = citationArray[4];
+		item.dateDecided = citationArray[1]
+		item.court = /Court(.+)/.exec(text(doc, "#co_docContentCourt"))[1];
+		item.abstractNote = "";
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "case";
+		trans.doWeb(doc, url);
+	});
+}
+
+function scrapeStatute(doc, url) {
+	const translator = Zotero.loadTranslator('web');
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+
+	translator.setHandler('itemDone', function (obj, item) {
+		const fullTitle = text(doc, "#co_documentTitle");
+		item.date = /(\D+) (\d+)/.exec(fullTitle)[2];
+		item.title = /(\D+) (\d+)/.exec(fullTitle)[1];
+		item.abstractNote = "";
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "statute";
+		trans.doWeb(doc, url);
+	});
+}
+
+function scrapeStatuteSection(doc, url) {
+	const translator = Zotero.loadTranslator('web');
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+
+	translator.setHandler('itemDone', function (obj, item) {
+		const fullTitle = /([^\n]+)/.exec(text(doc, ".co_title, .noTOC"))[0];
+		item.date = /(\D+) (\d+)/.exec(fullTitle)[2];
+		item.title = /(\D+) (\d+)/.exec(fullTitle)[1];
+		item.abstractNote = "";
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "statute";
+		trans.doWeb(doc, url);
+	});
+}
+
+
+function parseCitationList(citList) {
+	const allCitations = citList.match(/[\[|(](\d+)[\]|)] (\d*) ?([A-z. ]+) (\w[^Judgment])+/g); //annoyingly matchAll is not supported
+	if (allCitations.length === 1){ //if there is only one citation we'll have to use it
+		return /[\[|(](\d+)[\]|)] (\d*) ?([A-z. ]+) (\w[^Judgment])+/g.exec(allCitations[0]);
+	} else { //if there's more than one we will find the first one that isn't WLUK
+		for (const citation of allCitations) {
+			const citationAsArray = /[\[|(](\d+)[\]|)] (\d*) ?([A-z. ]+) (\w[^Judgment])+/g.exec(citation);
+			if (citationAsArray[3] !== "WLUK"){
+				return citationAsArray;
+			}
+		}
+	}
```

Comment:
Not having `matchAll` is indeed annoying, but there's a better way!

```suggestion
	let citationRe = /[\[|(](\d+)[\]|)] (\d*) ?([A-z. ]+) (\w[^Judgment])+/g;
	let citationAsArray;
	while ((citationAsArray = citationRe.exec(citation))) {
		if (citationAsArray[3] !== "WLUK") {
			return citationAsArray;
		}
	}
	return citationAsArray;
```

If there's no non-WLUK citation, do we want not to return anything at all? Or should we fall back on, say, the last citation? I assumed the latter in the code above.

Reply:
I had no idea you could do that with `exec()`! How fun is that!!

This approach would work well - the WLUK citations are the least ideal and cases that have only a WLUK citation are luckily few and far between, but a WLUK citation is better than no citation at all.